airbyte-ci/connectors/pipelines/tests/utils.py

```diff
@@ -13,7 +13,7 @@ def pick_a_random_connector(
     language: ConnectorLanguage = None, support_level: str = None, other_picked_connectors: list = None
 ) -> Connector:
     """Pick a random connector from the list of all connectors."""
-    all_connectors = list(ALL_CONNECTORS)
+    all_connectors = [c for c in list(ALL_CONNECTORS) if c.metadata is not None]
```

Comment:
❓Why this change?

Reply:
Some connector do not have metadata:
`airbyte-pagerduty-source, airbyte-harness-source, airbyte-victorops-source, airbyte-jenkins-source, streamr-airbyte-connector`

I want to avoid picking them randomly as this function does.

Reply:
I know were under a time crunch, but I think in this case we should do it right at the source.

These types of patches are where edge cases thrive

(Im sorry I know your under a time crunch 😢 )

Reply:
@bnchrch to follow up on your own comment here :P is this something we still want to address separately or think it's ok to keep around?

Reply:
Love that I ignored my own comment. Will be updating this

Reply:
I pushed a small commit to move this workaround into connector ops with a "HACK label"

and added a PR targeting this branch to fix the connector ops code
https://github.com/airbytehq/airbyte/pull/31166